### ISSUE TITLE
Performance optimization

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge-ruby#82]
**Expected behavior**
The performance of the ruby language runner must be comparable to js/python language runners

**Actual behavior**
The performance of gauge-ruby on Linux OS is slow

**Steps to replicate**
* project - https://github.com/sguptatw/gauge-ruby-performance-usages.git
* execute project `gauge run` from command line
> The performance of the runner is slow compared to js/python language runner 

**Version**
```
Gauge version: 0.9.9.nightly-2018-05-24
Commit Hash: 2150e31

Plugins
-------
ruby (0.5.1.nightly-2018-05-23)
```

**Comment by sguptatw** [comment]
The performance of ruby runner on Linux should be comparable with its performance on OSx

**Comment by sriv** [comment]
Setting `TCP_NODELAY` helps speed up execution in linux. Fix should be available in nightly >= 29/5/18

**Comment by sguptatw** [comment]
[Ruby Performance test suite](https://github.com/sguptatw/gauge-ruby-performance-usages.git) takes more than 5 minutes to execute where [Python Performance test suite] takes 5 seconds

Observed on VM Ubuntu
CPU - 2 cores
Memory - 2GB

**Comment by sriv** [comment]
This project has `Gemfile.lock` checked in, causing the `gauge-ruby` dependency to be fixed at revision `690b36679a5b3cd9941aaa73a716c8a56541693a` which is older than the revision that has the fix.

Run `bundle update` and then run `bundle exec gauge run specs` and see if it makes any difference.

**Comment by sguptatw** [comment]
@sriv Thanks. Works as expected after `bundle install`
